core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1522,6 +1522,17 @@ trait ZIOFunctions extends Serializable {
    * current fiber for this operation to return non-empty lists.
    */
   final def children: UIO[IndexedSeq[Fiber[_, _]]] = descriptor.flatMap(_.children)
+
+  /**
+   * Lifts a `ZIO[R, E, Reservation[R, E, A]]` into `ZManaged[R, E, A]`
```

Comment:
I would not mention the use of `ZManaged` here, because that's more advanced. Instead, I'd talk about how this differs from resource management with `bracket`: namely, this splits a bracket into an initial allocation, followed by an interruptible "acquire", followed by a terminal deallocation. You could mention this pattern is useful for concurrent data structures and other cases where the "deallocator" can tell if the allocation succeeded or not just by inspecting internal / external state.

If none of that makes sense, I can explain more and you can copy / paste it into the doc too. 😄